src/qibo/noise.py

```diff
@@ -0,0 +1,81 @@
+from qibo import gates
+
+class PauliError():
+    """Quantum error associated with the `qibo.core.gates.PauliNoiseChannel`.
+
+        Args:
+            options (tuple): see :class:`qibo.core.gates.PauliNoiseChannel`
+    """
+    def __init__(self, px=0, py=0, pz=0, seed=None):
+        self.options = px, py, pz, seed
+        self.channel = gates.PauliNoiseChannel
+
+
+class ThermalRelaxationError():
+    """Quantum error associated with the `qibo.core.gates.ThermalRelaxationChannel`.
+
+        Args:
+            options (tuple): see :class:`qibo.core.gates.ThermalRelaxationChannel`
+    """
+    def __init__(self, t1, t2, time, excited_population=0, seed=None):
+        self.options = t1, t2, time, excited_population, seed
+        self.channel = gates.ThermalRelaxationChannel
+
+
+class ResetError():
+    """Quantum error associated with the `qibo.core.gates.ResetChannel`.
+
+        Args:
+            options (tuple): see :class:`qibo.core.gates.ResetChannel`
+    """
+    def __init__(self, p0, p1, seed=None):
+        self.options = p0, p1, seed
+        self.channel = gates.ResetChannel
+
+
+class NoiseModel():
+    """Class for the implementation of a custom noise model."""
```

Comment:
I would add a short example here, such as the one you have in the first post, particularly to show the usage of `.add` and demonstrate the API more clearly (for example that the `gate` argument should be a class type, eg. `gates.X`, etc.).

Reply:
I agree. I've added the same example.